cli/internal/globby/globby.go

```diff
@@ -101,7 +101,13 @@ func globWalkFs(fsys iofs.FS, fsysRoot string, basePath string, includePatterns
 		// This will not error as it follows the call to checkRelativePath.
 		iofsRelativePath, _ := fs.IofsRelativePath(fsysRoot, excludePath)
 
-		// Excludes operate on entire folders.
+		// In case this is a file pattern and not a directory, add the exact pattern.
+		// In the event that the user has already specified /**,
+		if !strings.HasSuffix(iofsRelativePath, "**") {
```

Comment:
So we still add `/**` to everything, but in the event that your current thing does not end in `**` we _also_ add it to the exclusions without appending `/**`.

I'd love to get tests constraining the behavior for more situations:
- `**/` => gets added as `**/` and `**/**`?
- `foo**` => gets added as `foo**/**` => is this a gap in the per-file exclusion?
- `**foo` => gets added as `**foo`, `**foo/**`

Reply:
> `foo**` => gets added as `foo**/**` => is this a gap in the per-file exclusion?
Good catch. I updated the suffix that we check for from `**` -> `/**`. Paths are already required to be in posix format.

Reply:
Paths are apparently not required to be in posix format. Now using the appropriate system-specific suffix to look for.